tests/e2e/specs/shopper/checkout-payments.test.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * External dependencies
+ */
+import {
+	merchant,
+	setCheckbox,
+	settingsPageSaveChanges,
+	verifyCheckboxIsSet,
+} from '@woocommerce/e2e-utils';
+
+/**
+ * Internal dependencies
+ */
+import {
+	getNormalPagePermalink,
+	shopper,
+	visitPostOfType,
+} from '../../../utils';
+
+const block = {
+	name: 'Checkout',
+};
+
+const simpleProductName = 'Woo Single #1';
+
+if ( process.env.WOOCOMMERCE_BLOCKS_PHASE < 2 )
+	// eslint-disable-next-line jest/no-focused-tests
+	test.only( `skipping ${ block.name } tests`, () => {} );
+
+describe( `${ block.name } Block (frontend)`, () => {
+	let productPermalink;
+
+	beforeAll( async () => {
+		// prevent CartCheckoutCompatibilityNotice from appearing
+		await page.evaluate( () => {
+			localStorage.setItem(
+				'wc-blocks_dismissed_compatibility_notices',
+				'["checkout"]'
+			);
+		} );
+		await merchant.login();
+
+		// Get product page permalink.
+		await visitPostOfType( simpleProductName, 'product' );
+		productPermalink = await getNormalPagePermalink();
+
+		await merchant.logout();
+	} );
+
+	afterAll( async () => {
+		// empty cart from shortcode page
+		await shopper.goToCart();
+		await shopper.removeFromCart( simpleProductName );
+		await page.evaluate( () => {
+			localStorage.removeItem(
+				'wc-blocks_dismissed_compatibility_notices'
+			);
+		} );
+	} );
+
+	it( 'allows customer to choose available payment methods', async () => {
+		await shopper.goToCheckoutBlock();
```

Comment:
This line doesn't seem necessary! We go to the Checkout Block after adding an item to our cart. So, instead of:

```js
await shopper.goToCheckoutBlock();
await page.goto( productPermalink );
await shopper.addToCart();
await shopper.goToCheckoutBlock();
```

We can have:

```js
await page.goto( productPermalink );
await shopper.addToCart();
await shopper.goToCheckoutBlock();
```

Reply:
I cannot explain what's going on, but when I remove the first `await shopper.goToCheckoutBlock();` call, the test keeps failing. I only kept the first call, to avoid flakiness. 😬

Reply:
Might have been a problem due to repeatable reads. When spinning up a fresh e2e environment,  the tests runs expected. 😀